connectivity/FEATURE_BLE/source/generic/GapImpl.cpp

```diff
@@ -1444,8 +1450,11 @@ ble_error_t Gap::update_ll_address_resolution_setting()
         }
 #endif // BLE_ROLE_OBSERVER
     }
-
+#if BLE_GAP_HOST_BASED_PRIVACY
```

Comment:
Can it be return early or execute the body ? Computations above are pointless if host base privacy is enabled.

Reply:
narrowed down the meaning to address resolution and fixed ifdefs accordingly